pkgs/tools/package-management/nix/common.nix

```diff
@@ -277,6 +278,12 @@ self = stdenv.mkDerivation {
         fi
         touch $out
       '';
+
+      /** Intended to test `lib`, but also a good smoke test for Nix */
+      nixpkgs-lib = import ../../../../lib/tests/test-with-nix.nix {
+        inherit lib pkgs;
+        nix = self;
+      };
```

Comment:
See also this:

https://github.com/NixOS/nixpkgs/blob/2b875be1ab298154c70f066b6c5f30abf0369d7c/lib/tests/release.nix#L11-L17

I like that you can now test Nix with Nix against every Nix.